eng/native/configuretools.cmake

```
@@ -7,13 +7,13 @@ if (CMAKE_C_COMPILER MATCHES "-?[0-9]+(\.[0-9]+)?$")
 endif()
 
 if(NOT WIN32 AND NOT CLR_CMAKE_TARGET_BROWSER)
-  if(CMAKE_CXX_COMPILER_ID MATCHES "Clang")
+  if(CMAKE_C_COMPILER_ID MATCHES "Clang")
     if(APPLE)
       set(TOOLSET_PREFIX "")
     else()
       set(TOOLSET_PREFIX "llvm-")
     endif()
-  elseif(CMAKE_CXX_COMPILER_ID MATCHES "GNU")
+  elseif(CMAKE_C_COMPILER_ID MATCHES "GNU")
     if(CMAKE_CROSSCOMPILING)
       set(TOOLSET_PREFIX "${CMAKE_CXX_COMPILER_TARGET}-")
```

This should also be changed to `LANG:C`:
```suggestion
      set(TOOLSET_PREFIX "${CMAKE_C_COMPILER_TARGET}-")
```